libbeat/processors/add_docker_metadata/config.go

```diff
@@ -6,23 +6,26 @@ import (
 	"github.com/elastic/beats/libbeat/common/docker"
 )
 
-// Config for docker processor
+// Config for docker processor.
 type Config struct {
-	Host        string            `config:"host"`
-	TLS         *docker.TLSConfig `config:"ssl"`
-	Fields      []string          `config:"match_fields"`
-	MatchSource bool              `config:"match_source"`
-	SourceIndex int               `config:"match_source_index"`
+	Host        string            `config:"host"`               // Docker socket (UNIX or TCP socket).
+	TLS         *docker.TLSConfig `config:"ssl"`                // TLS settings for connecting to Docker.
+	Fields      []string          `config:"match_fields"`       // A list of fields to match a container ID.
+	MatchSource bool              `config:"match_source"`       // Match container ID from a log path present in source field.
+	SourceIndex int               `config:"match_source_index"` // Index in the source path split by / to look for container ID.
+	MatchPIDs   []string          `config:"match_pids"`         // A list of fields containing process IDs (PIDs).
+	HostFS      string            `config:"system.hostfs"`      // Specifies the mount point of the host’s filesystem for use in monitoring a host from within a container.
 
-	// Annotations are kept after container is killled, until they haven't been accessed
-	// for a full `cleanup_timeout`:
+	// Annotations are kept after container is killed, until they haven't been
+	// accessed for a full `cleanup_timeout`:
 	CleanupTimeout time.Duration `config:"cleanup_timeout"`
 }
 
 func defaultConfig() Config {
 	return Config{
 		Host:        "unix:///var/run/docker.sock",
 		MatchSource: true,
-		SourceIndex: 4,
+		SourceIndex: 4, // Use 4 to match the CID in /var/lib/docker/containers/<container_id>/*.log.
+		MatchPIDs:   []string{"process.pid", "process.ppid"},
```

Comment:
Perhaps we can add a default for `hostFS` here?

Reply:
In the cgroup package if the param is empty it will set a default. If you want me to add one here (`/`) to override the package default (also `/`) I can add that.

Reply:
nah, that's ok! I didn't know `""` was a valid default, works for me :+1: